swim/member.go

```diff
@@ -122,10 +123,14 @@ func (l LabelMap) checksum() int32 {
 	// make sure the buffer is empty after getting a buffer from our pool
 	lb.Reset()
 	for key, value := range l {
-		// decide if we want to escape both fields to enforce uniqueness of the
-		// key-value checksum.
+		// The bytes used for checksumming are the following:
+		// <4 bytes: length of key>
+		binary.Write(lb, binary.BigEndian, int32(len(key)))
```

Comment:
Note: `len(string)` returns the length in _bytes_. This means that in node we should use [`Buffer.byteLength`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_bytelength_string_encoding) instead of `string.length`